### PR TITLE
add mkdocs

### LINKED ISSUE #68: add mkdocs setup
like titiler? https://github.com/developmentseed/titiler

### PR DESCRIPTION
**Pull request by vincentsarago**
closes #68 

This PR add Mkdocs documentation + doc building gh action

cc @geospatial-jeff

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    logo: 'material/home'
+    repo: 'fontawesome/brands/github'
+  name: 'material'
+  language: 'en'
+  palette:
+    primary: 'blue'
+    accent:  'light blue'
+  font:
+    text: 'Nunito Sans'
+    code: 'Fira Code'
```
Comment (vincentsarago) [inline mkdocs.yml:39]: Let me change this to match Arturo colors

File: .github/workflows/cicd.yaml [hunk 2]
```diff
@@ -40,18 +40,14 @@ jobs:
         with:
           python-version: "3.8"
 
+      - name: Lint code
+        uses: pre-commit/action@v2.0.0
```
Comment (kylebarron) [inline .github/workflows/cicd.yaml]: 👀

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -1,35 +1,34 @@
...
         hooks:
             - id: flake8
-              language_version: python3.7
+              language_version: python3.8
               args: [
                   # E501 let black handle all line length decisions
                   # W503 black conflicts with "line break before operator" rule
                   # E203 black conflicts with "whitespace before ':'" rule
-                  '--ignore=E501,W503,E203']
+                  '--ignore=E501,W503,E203,C901']
```
Comment (kylebarron) [inline .pre-commit-config.yaml]: Add a comment for what `C901` is?

File: CHANGES.md [hunk 4]
```diff
@@ -1,4 +1,13 @@
 
+## 1.1.0 (2021-01-28)
+
+* Improve how the library declares API extensions (https://github.com/arturo-ai/arturo-stac-api/pull/54)
+* Add postgres bulk transactions client (https://github.com/arturo-ai/arturo-stac-api/pull/59)
+* Update TiTiler version (https://github.com/arturo-ai/arturo-stac-api/pull/61)
+* Use attrs instead of dataclasses (https://github.com/arturo-ai/arturo-stac-api/pull/73)
+* Remove postgres database connection from API layer (https://github.com/arturo-ai/arturo-stac-api/pull/74)
+* Fix precommit config (https://github.com/arturo-ai/arturo-stac-api/pull/75)
```
Comment (kylebarron) [inline CHANGES.md:9]:
```suggestion
* Fix `pre-commit` config (https://github.com/arturo-ai/arturo-stac-api/pull/75)
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -10,7 +10,7 @@ $ cd arturo-stac-api
 $ pip install -e .[dev]
 ```
 
-**Python3.7 only**
+**Python3.8 only**
```
Comment (kylebarron) [inline CONTRIBUTING.md:13]: This means `pre-commit` won't work anymore if your global Python is 3.7?

File: Dockerfile [hunk 6]
```diff
@@ -20,7 +20,7 @@ COPY . ./
 
 ENV APP_HOST=0.0.0.0
 ENV APP_PORT=80
-ENV RELOAD=""
+ENV RELOAD="true"
```
Comment (kylebarron) [inline Dockerfile]: Do you need to merge latest master? This shouldn't be a diff https://github.com/arturo-ai/arturo-stac-api/blob/master/Dockerfile#L23

File: docs/stylesheets/extra.css [hunk 7]
```diff
@@ -0,0 +1,3 @@
+:root {
+  --md-primary-fg-color:        rgb(18, 97, 141);
```
Comment (kylebarron) [inline docs/stylesheets/extra.css:2]: You didn't like any of the default colors? https://squidfunk.github.io/mkdocs-material/setup/changing-the-colors/#primary-color

**Review by kylebarron — APPROVED** [review]

**Comment by kylebarron** [comment]
Weird... I guess Github didn't show me the accurate diff?

File: README.md [hunk 8]
```diff
@@ -1,4 +1,26 @@
-# arturo-stac-api ![arturo-stac-api](https://github.com/arturo-ai/arturo-stac-api/workflows/arturo-stac-api/badge.svg)
```
Comment (kylebarron) [inline README.md:1]: You probably still want
```
# arturo-stac-api
```
Right now there's no title

File: Dockerfile (hunk 6 above)
Comment (vincentsarago) [reply Dockerfile]: yeah I did after I started this PR 🤦

File: docs/stylesheets/extra.css (hunk 7 above)
Comment (vincentsarago) [reply docs/stylesheets/extra.css:2]: used the color from Arturo web site 🤷‍♂️

File: CONTRIBUTING.md (hunk 5 above)
Comment (vincentsarago) [reply CONTRIBUTING.md:13]: the language in pre-commit is set to python 3.8 so I guess it's just better to use python 3.8 🤷‍♂️

File: CHANGES.md [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+* Improve how the library declares API extensions (https://github.com/arturo-ai/arturo-stac-api/pull/54)
+* Add postgres bulk transactions client (https://github.com/arturo-ai/arturo-stac-api/pull/59)
+* Update TiTiler version (https://github.com/arturo-ai/arturo-stac-api/pull/61)
+* Use attrs instead of dataclasses (https://github.com/arturo-ai/arturo-stac-api/pull/73)
+* Remove postgres database connection from API layer (https://github.com/arturo-ai/arturo-stac-api/pull/74)
+* Fix `pre-commit` config (https://github.com/arturo-ai/arturo-stac-api/pull/75)
+
+## 1.0.0 (2020-09-25)
+
+* First PyPi release!
```
Comment (vincentsarago) [inline CHANGES.md:13]: I can revert to text if you don't like md

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+  stac_api.errors \
+  stac_api.config \
+  stac_api.models.database \
+  stac_api.models.decompose \
+  stac_api.models.links \
+  stac_api.models.ogc \
+  stac_api.models.schemas \
+  stac_api.api.routes \
+  stac_api.api.models \
+  stac_api.api.app
```
Comment (vincentsarago) [inline CONTRIBUTING.md:53]: update this!

File: docs/stylesheets/extra.css (hunk 7 above)
Comment (geospatial-jeff) [inline docs/stylesheets/extra.css:2]: you were close!
```suggestion
  --md-primary-fg-color:        rgb(13, 118, 160);
```

File: CHANGES.md (hunk 9 above)
Comment (geospatial-jeff) [reply CHANGES.md:13]: no this is great!

File: .github/workflows/deploy_mkdocs.yml [hunk 11]
```diff
@@ -0,0 +1,51 @@
+name: Publish docs via GitHub Pages
+
+on:
+  push:
+    branches:
+      - master
+    paths:
+      # Only rebuild website when docs have changed
```
Comment (kylebarron) [inline .github/workflows/deploy_mkdocs.yml:8]: Just a note that when you also display API docs, it might be worthwhile to rebuild the website more often.

**Pull request merged**